hometask-3

MakarovK opened the pull request:
1. Добавлены зависимости io.swagger.core.v и org.projectlombok в родительский pom и наследникам delivery-service, order-service.
2. Реализованы классы DTO для delivery-service, order-service.
3. Реализованы REST контролле для delivery-service, order-service.

MakarovK committed:
1. Добавлены зависимости io.swagger.core.v и org.projectlombok в родительский pom и наследникам delivery-service, order-service.
2. Реализованы классы DTO для delivery-service, order-service.
3. Реализованы REST контролле для delivery-service, order-service.

MakarovK committed:
1. Добавлены зависимости io.swagger.core.v и org.projectlombok в роди…
…тельский pom и наследникам delivery-service, order-service.

2. Реализованы классы DTO для delivery-service, order-service.
3. Реализованы REST контролле для delivery-service, order-service.

MakarovK committed:
1. Добавлены зависимости io.swagger.core.v и org.projectlombok в родительский pom и наследникам delivery-service, order-service.
2. Реализованы классы DTO для delivery-service, order-service.
3. Реализованы REST контролле для delivery-service, order-service.

MakarovK committed:
Merge branch 'hometask-3' of https://github.com/MakarovK/ru-liga-food-service into hometask-3

Zero-imagination commented on delivery-service/src/main/java/ru/liga/controller/DeliveryController.java:
```diff
@@ -0,0 +1,58 @@
+    @Operation (summary = "Получить детали доставки по ID")
+    @GetMapping("/{id}")
+    public DeliveryDTO getDeliveryById(@PathVariable("id") Long id,
+                                       @PathVariable("restaurantAddress") String restaurantAddress,
```
здесь будет ругаться java, так как pathVariable ты указал только id (на 15 строчке которое), остальные 4 параметрам неоткуда взяться

MakarovK committed:
1. Добавлены новые классы в DTO.
2. Реализован синглтон для проверки контроллеров.
3. Реализованы GET, POST контроллеры вместо PUT контроллера затычка.
4. Так же реализован GET в delivery.
5. Всё разобрано, скомпилировано и протестировано в Postman.

Zero-imagination commented on delivery-service/pom.xml:
```diff
@@ -27,6 +27,20 @@
             <groupId>org.yaml</groupId>
             <artifactId>snakeyaml</artifactId>
```
для каждого модуля можно не указывать shakeyaml, достаточно только в dependency_bom, так как он имеет транзитивную зависимость

Zero-imagination commented on order-service/src/main/java/ru/liga/dto/orders/OrdersDTO.java:
```diff
@@ -0,0 +1,70 @@
+    private OrdersDTO() {
+
+        orderDTOList = new ArrayList<>();
+        orderDTOList.add(new OrderDTO()
```
а зачем инициализируешь здесь?

Zero-imagination commented on order-service/src/main/java/ru/liga/dto/orders/OrdersDTO.java:
```diff
@@ -0,0 +1,70 @@
+@Data
+@Accessors(chain = true)
+@Schema(description = "Синглтон для проверки в Postman")
+public class OrdersDTO {
```
этот класс используется для контроллера? может тогда стоит назвать его OrderResponse, или OrderRequest.
обычно в классах не пишут множественное число

Zero-imagination commented on order-service/src/main/java/ru/liga/dto/orders/RestaurantDTO.java:
```diff
@@ -0,0 +1,17 @@
+    public RestaurantDTO() {
+    }
+
+    public RestaurantDTO(String name) {
```
можно использовать аннотации ломбока для конструктора. requiredArgConstructor. Тогда не придется писать их руками
